### ISSUE TITLE
Warning on local variable names masking global variable names

### ISSUE THREAD

**Issue by sdwarwick** [golang/lint#255]
Golint doesn't flag on such global / local overloading conditions.  In the following case, it would have been a great help in finding a runtime error.   
```
package main

var myIO   *io.Writer

function main() {
.
.
myIO := (create writer)    // error - created a local variable rather than setting 
                           // the global with a := rather than =
.
}

function useit ()
{
.
.
fmt.Fprint(myIO, "stuff)    // null pointer at runtime since myIO defined, but set to null
.
}
```

Flagging global/local variable overloading would seem like a natural for the linter!

**Comment by sdwarwick** [comment]
Go tools vet has an implementation of this but is normally switched off.
